### PR TITLE
Adapt onboarding information

### PR DESCRIPTION
**Pull request by marcushoepfner**
Issue #675

Re-read onboarding information on start up (similar to action sets) and update the information

### CODE REVIEW

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java [hunk 1]
```diff
@@ -2860,7 +2881,7 @@ private void addActionSet(MPerspective perspective, MPerspective temporary) {
 		List<String> tags = perspective.getTags();
 		List<String> extendedTags = temporary.getTags();
 		for (String extendedTag : extendedTags) {
-			if (!tags.contains(extendedTag)) {
+			if (extendedTag.startsWith(ModeledPageLayout.ACTION_SET_TAG) && !tags.contains(extendedTag)) {
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java:2886]: why is that needed?

**Review by vogella — APPROVED** [review]
Thanks @marcushoepfner works fine. The test failures seem unrelated.

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java [hunk 2]
```diff
@@ -2826,14 +2828,17 @@ private void extendPerspectives(MPerspectiveStack perspectiveStack) {
 						desc, this, true);
 
 				PerspectiveExtensionReader reader = new PerspectiveExtensionReader();
-				reader.setIncludeOnlyTags(new String[] { IWorkbenchRegistryConstants.TAG_ACTION_SET });
+				reader.setIncludeOnlyTags(new String[] { IWorkbenchRegistryConstants.TAG_ACTION_SET,
+						IWorkbenchRegistryConstants.TAG_EDITOR_ONBOARDING_COMMAND });
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java]: Why do you only add the "..command" here? what if e.g. on the text changes?

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java [hunk 3]
```diff
@@ -66,6 +66,9 @@
 
 public class ModeledPageLayout implements IPageLayout {
 
+	/**
```
Comment (BeckerWdf) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java]: what about this empty block?

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java (hunk 1 above)
Comment (marcushoepfner) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java:2886]: This method handles action sets actually. Before my commit, there was only handling for action sets. refer to method extendPerspectives "includeTags".
Since there is also handling for onboarding now, this method need to be restricted now.

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java (hunk 2 above)
Comment (marcushoepfner) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchPage.java]: Added text and image as well.
I needed to change PerspectiveExtensionReader.processExtension(IConfigurationElement element) for that.
Please have a look.

Tests will follow soon with a new PR, like discussed already.

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java (hunk 3 above)
Comment (marcushoepfner) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java]: removed

**Review by BeckerWdf — APPROVED** [review]
